File: Python/Arrays & Hashing/encode-and-decode-strings.py
class Solution:
    def encode(self, strs):
        text = ''
        for string in strs:
            text += str(len(string)) + '#' + string
        return text

    def decode(self, s):
        result, i = [], 0

        while i < len(s):
            mark = s.index('#', i)
            length = int(s[i : mark])
            result.append(s[mark + 1 : mark + length + 1])
            i = mark + length + 1

        return result


s = Solution().encode(["we","say",":","yes","!@#$%^&*()"])
print(["we","say",":","yes","!@#$%^&*()"])
print(Solution().decode(s))

File: Cpp/Arrays & Hashing/encode-and-decode-strings.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string result = "";
        for (string str : strs) {
            result += to_string(str.length()) + "#" + str;
        }
        return result;
    }

    vector<string> decode(string s) {
        vector<string> result;
        int i = 0;
        
        while (i < s.length()) {
            int mark = s.find('#', i);
            int length = stoi(s.substr(i, mark - i));
            result.push_back(s.substr(mark + 1, length));
            i = mark + length + 1;
        }

        return result;
    }
};
